src/index.js

```diff
@@ -73,14 +81,11 @@ class MySqlConnection {
   }
 
   getConnection() {
-    if (this.connection) return Promise.resolve(this.connection);
```

Comment:
Are you saying we shouldn't reuse the previously acquired connection and should ask for a new one from the pool every time getConnection is called?

Reply:
Yes. :)